integrated_channels/api/v1/canvas/views.py

```diff
@@ -0,0 +1,53 @@
+"""
+Viewsets for Canvas
+"""
+from rest_framework import permissions, status, viewsets
+from rest_framework.response import Response
+from edx_rbac.decorators import permission_required
+from enterprise.models import EnterpriseCustomerUser
+
+
+from integrated_channels.canvas.models import CanvasEnterpriseCustomerConfiguration
+
+from .serializers import CanvasEnterpriseCustomerConfigurationSerializer
+
+
+class CanvasConfigurationViewSet(viewsets.ModelViewSet):
+    serializer_class = CanvasEnterpriseCustomerConfigurationSerializer
+    permission_classes = (permissions.IsAuthenticated, permissions.DjangoModelPermissions,)
+
+    def get_queryset(self):
+        if self.request.user.is_staff:
+            return CanvasEnterpriseCustomerConfiguration.objects.all()
+        try:
+            enterprise_customer = EnterpriseCustomerUser.objects.get(user_id=self.request.user.id).enterprise_customer
+            return CanvasEnterpriseCustomerConfiguration.objects.filter(
+                enterprise_customer=enterprise_customer
+            )
+        except EnterpriseCustomerUser.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+    @permission_required('enterprise.can_access_admin_dashboard')
+    def list(self, request):
+        return super(CanvasConfigurationViewSet, self).list(request)
+
+    @permission_required('enterprise.can_access_admin_dashboard')
+    def create(self, request):
+        return super(CanvasConfigurationViewSet, self).create(request)
+
+    @permission_required('enterprise.can_access_admin_dashboard')
+    def retrieve(self, request, pk=None):
+        return super(CanvasConfigurationViewSet, self).retrieve(request, pk=pk)
```

Comment:
interesting. I haven't seen this many methods defined yet. How do the endpoints differentiate retrieve vs list? I get partial_update == PATCH vs update/create == POST. Does Django just know based on whether you pass a parameter or not?

Reply:
Does it based on the endpoint requested and which HTTP verb you use. So:

* list -> `GET api/v1/resource/`
* retrieve -> `GET api/v1/resource/<id>`
* create -> `POST api/v1/resource/`
* update -> `PUT api/v1/resource/<id>`
* partial_update -> `PATCH api/v1/resource/<id>`
* destroy -> `DELETE api/v1/resource/<id>`